packages/solutions/module-tools/src/config/merge.ts

```diff
@@ -113,6 +113,8 @@ export const mergeDefaultBaseConfig = async (
 
   const esbuildOptions = pConfig.esbuildOptions ?? defaultConfig.esbuildOptions;
   return {
+    shims: pConfig.shims ?? defaultConfig.shims,
+    autoExtension: pConfig.autoExtension ?? defaultConfig.autoExtension,
```

Comment:
New configs need to be documented

Reply:
If this pr is clear, I will add it to document.

Reply:
ok~

Reply:
or you can release a canary version and test this feature in the Rsbuild repo first